contracts/Lockbox.cdc

```diff
@@ -0,0 +1,130 @@
+import FlowToken from 0x0ae53cb6e3f42a79
+import FungibleToken from 0xee82856bf20e2aa6
+
+pub let UNLOCKED = "UNLOCKED"
+
+pub contract Lockbox {
+
+  pub resource TokenBucket {
+    
+    pub var vault: @FlowToken.Vault
+
+    pub var withdrawalLimit: UFix64
+
+    pub fun setWithdrawalLimit(limit: UFix64) {
```

Comment:
Does this open up the contract to a variant of the ERC20Multiple Withdrawal Attack?

https://blockchain-projects.readthedocs.io/multiple_withdrawal.html

If so, and if we do not want to change the interface, the mitigation would be to use two transactions to change the withdrawal limit: one to set the limit to zero, then another to set it to its new value.